test/Maestro.DataProviders.Tests/ConfigurationIngestionTests.cs

```diff
@@ -0,0 +1,1141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using AwesomeAssertions;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.DataProviders.ConfigurationIngestion;
+using Microsoft.DotNet.MaestroConfiguration.Client.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using ProductConstructionService.Api.Tests;
+
+namespace Maestro.DataProviders.Tests;
+
+[TestFixture]
+public class ConfigurationIngestorTests
+{
+    private TestDatabase _testDatabase = null!;
+    private BuildAssetRegistryContext _context = null!;
+    private IConfigurationIngestor _ingestor = null!;
+    private string _testNamespace = string.Empty;
+
+    [SetUp]
+    public async Task SetUp()
+    {
+        _testNamespace = "test-namespace-" + Guid.NewGuid();
+
+        // TODO: This still creates a whole DB per test which is super slow
+        //       We should only create the DB per test suite via a OneTimeSetUp and OneTimeTearDown
+        _testDatabase = new TestDatabase("TestDB_Ingestion_" + TestContext.CurrentContext.Test.MethodName);
+        var connectionString = await _testDatabase.GetConnectionString();
+
+        var options = new DbContextOptionsBuilder<BuildAssetRegistryContext>()
+            .UseSqlServer(connectionString)
+            .Options;
+
+        _context = new BuildAssetRegistryContext(options);
+
+        var services = new ServiceCollection()
+            .AddSingleton(_context)
+            .AddSingleton<ISqlBarClient>(new SqlBarClient(_context, null))
+            .AddConfigurationIngestion();
+
+        _ingestor = services.BuildServiceProvider()
+            .GetRequiredService<IConfigurationIngestor>();
+
+        // Pre-populate repositories used in tests
+        await EnsureRepositoriesExist(
+            "https://github.com/dotnet/runtime",
+            "https://github.com/dotnet/runtime2",
+            "https://github.com/dotnet/runtime3",
+            "https://github.com/dotnet/aspnetcore",
+            "https://github.com/dotnet/old",
+            "https://github.com/dotnet/target");
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        _context?.Dispose();
+        _testDatabase?.Dispose();
+    }
+
+    #region Empty Database Tests
+
+    [Test]
+    public async Task IngestConfigurationAsync_EmptyDatabase_CreatesAllEntities()
+    {
+        // Arrange
+        var configData = CreateBasicConfigurationData();
+
+        // Act
+        var result = await _ingestor.IngestConfigurationAsync(configData, _testNamespace);
+
+        // Assert
+        Assert.That(result.Channels.Creations.Count, Is.EqualTo(1));
```

Comment:
this is still not using AwesomeAssertions, should we change all of them to use it?